autoagora/logs_db.py

```diff
@@ -27,9 +51,96 @@ def return_query_body(self, query):
         query = "query " + graphql.print_ast(query)
         return query
 
-    async def get_most_frequent_queries(
+    async def get_most_frequent_queries_null_time(
         self, subgraph_ipfs_hash: str, min_count: int = 100
     ):
+        async with self.pgpool.acquire() as connection:
+            rows = await connection.fetch(
+                """
+                SELECT
+                    hash,
+                    query
+                    
+                FROM
+                    query_skeletons
+                INNER JOIN
+                (
+                    SELECT
+                        query_hash as qhash,
+                        count(id) as count_id
+                    FROM
+                        query_logs
+                    WHERE
+                        subgraph = $1
+                        AND query_time_ms IS NULL
+                    GROUP BY
+                        qhash
+                    HAVING
+                        Count(id) >= $2
+                ) as query_logs
+                ON
+                    qhash = hash
+                ORDER BY
+                    count_id DESC
+                """,
+                subgraph_ipfs_hash,
+                min_count,
+            )
+            return [
+                LogsDB.MRQ_Info(
+                    hash=row[0],
+                    query=self.return_query_body(row[1])
+                    if self.return_query_body(row[1])
+                    else "null",
+                )
+                for row in rows
+            ]
+
+    async def get_query_variables_from_query_hash(self, hash):
+        async with self.pgpool.acquire() as connection:
+            query_variables = await connection.fetch(
+                """
+                SELECT
+                    query_variables
+                FROM 
+                    query_logs
+	            WHERE
+                    query_hash = $1;
+                """,
+                hash,
+            )
+            return query_variables
```

Comment:
I think it'd be simpler to directly ask the DB for a single random entry, instead of getting them all. Also helps with the performance / memory usage of AA.

Reply:
youre right will make this

Reply:
As we discussed, asking for a random entry using SQL is not great for DB performance. So another solution has been worked out. Closing.